app/src/main/java/com/linminitools/myrsync/PluginBundleValues.java

```diff
@@ -0,0 +1,120 @@
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import static com.linminitools.myrsync.myRsyncApplication.configs;
+import static com.twofortyfouram.assertion.Assertions.assertNotEmpty;
+import static com.twofortyfouram.assertion.Assertions.assertNotNull;
+
+/**
+ * Manages the {@link com.twofortyfouram.locale.api.Intent#EXTRA_BUNDLE EXTRA_BUNDLE} for this
+ * plug-in.
+ */
+@ThreadSafe
+public final class PluginBundleValues {
+
+    /**
+     * Type: {@code String}.
+     * <p>
+     * String message to display in a Toast message.
+     */
+    @NonNull
+    public static final String BUNDLE_EXTRA_INT_CONFIG
+            = "com.linminitools.myrsync.INT_CONFIG"; //$NON-NLS-1$
+
+    /**
+     * Type: {@code int}.
+     * <p>
+     * versionCode of the plug-in that saved the Bundle.
+     */
+    /*
+     * This extra is not strictly required, however it makes backward and forward compatibility
+     * significantly easier. For example, suppose a bug is found in how some version of the plug-in
+     * stored its Bundle. By having the version, the plug-in can better detect when such bugs occur.
+     */
+    @NonNull
+    public static final String BUNDLE_EXTRA_INT_VERSION_CODE
+            = "com.linminitools.myrsync.INT_VERSION_CODE"; //$NON-NLS-1$
+
+    /**
+     * Method to verify the content of the bundle are correct.
+     * <p>
+     * This method will not mutate {@code bundle}.
+     *
+     * @param bundle bundle to verify. May be null, which will always return false.
+     * @return true if the Bundle is valid, false if the bundle is invalid.
+     */
+    public static boolean isBundleValid(@Nullable final Bundle bundle) {
+        if (null == bundle) {
+            return false;
+        }
+
+        try {
+            BundleAssertions.assertHasInt(bundle, BUNDLE_EXTRA_INT_CONFIG);
+            BundleAssertions.assertHasInt(bundle, BUNDLE_EXTRA_INT_VERSION_CODE);
+            BundleAssertions.assertKeyCount(bundle, 2);
+        } catch (final AssertionError e) {
+            Lumberjack.e("Bundle failed verification%s", e); //$NON-NLS-1$
+            return false;
+        }
+
+        int configId = bundle.getInt(BUNDLE_EXTRA_INT_CONFIG);
+
+        boolean configFound = false;
+
+        for (RS_Configuration c : configs) {
+            if (c.id == configId) {
+                configFound = true;
+                break;
+            }
+        }
+
+        return configFound;
+    }
+
+    /**
+     * @param context Application context.
+     * @param configId The id of the config file selected.
+     * @return A plug-in bundle.
+     */
+    @NonNull
+    public static Bundle generateBundle(@NonNull final Context context,
+                                        @NonNull final Integer configId) {
+        assertNotNull(context, "context"); //$NON-NLS-1$
+//        assertNotEmpty(configId, "configId"); //$NON-NLS-1$
```

Comment:
Maybe we can remove this?

Reply:
Ooh, don't think that was meant to be commented out!